### PR TITLE
PubMatic: Support for Outstream Renderer (BlueBilyWig Renderer)

### PR DESCRIPTION
**Pull request by pm-shashank-jain**
## Type of change
- [X] Feature

## Description of change
PubMatic adapter now supports bluebillywig renderer for outstream videos.

## Other information
PR for Updating Bidder Doc for new parameter -> https://github.com/prebid/prebid.github.io/pull/2170

### CODE REVIEW

**Comment by pm-shashank-jain** [comment]
Hi @jsnellbaker, video params are mandatory for video request in pubmatic adapter that's why it is failing. 
Please use this as an adUnit 
```
[{
                code: 'Div1',
                mediaTypes: {
                    video: {
                        playerSize: [640, 480],
                        context: 'outstream'
                    }
                },
                bids: [{
                    bidder: 'pubmatic',
                    params: {
                        adSlot: 'Div1',
                        publisherId: '5890',
                        outstreamAU: 'renderer_test_pubmatic',
						adSlot: 'Div1@640x480', //ad_id or tagid
						pmzoneid: 'zone1',
						lat: '40.712775  ',
						lon: '   -74.005973',
						yob: '  1982  ',
						kadpageurl: 'http://www.microsoft.com/key1=1&key2=2',							
						gender: '   M   ',
						kadfloor:'  1.75   ',
                        video: {
                            mimes: ['video/mp4', 'video/x-flv'],
                            skippable: true,
                            minduration: 5,
                            maxduration: 30,
							startdelay: 15,
							playbackmethod: [1, 3],
							api: [1, 2],
							protocols: [2, 3],
							w: 640,
							h: 480,
							battr: [13, 14],
							linearity: 1,
							placement: 2,
							minbitrate: 100,
							maxbitrate: 4096
                        }
                    }
                }]
            }];
```

**Comment by pm-shashank-jain** [comment]
@jsnellbaker can you please look at the same ? and please let me know if more information is required.
TIA

**Comment by jsnellbaker** [comment]
Hi @pm-shashank-jain 

Sorry for the delay, I was ooo for a bit.  Thank you for providing the test adUnit.  I tried to use it, but I wasn't seeing a response from the adserver.  Below is a copy of the request:
```
{"id":"1597079225765","at":1,"cur":["USD"],"imp":[{"id":"2919668b346c4","tagid":"Div1","bidfloor":1.75,"secure":1,"ext":{"pmZoneId":"zone1"},"bidfloorcur":"USD","video":{"mimes":["video/mp4","video/x-flv"],"minduration":5,"maxduration":30,"startdelay":15,"playbackmethod":[1,3],"api":[1,2],"protocols":[2,3],"w":640,"h":480,"battr":[13,14],"linearity":1,"placement":2,"minbitrate":100,"maxbitrate":4096,"ext":{"video_skippable":1}}}],"site":{"page":"http://www.microsoft.com/key1=1&key2=2","ref":"http://test.localhost:9999/integrationExamples/gpt/outstream.html?pbjs_debug=true","publisher":{"id":"5890"},"domain":"www.microsoft.com"},"device":{"ua":"Mozilla/5.0 (Macintosh; Intel Mac OS X 10_14_6) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/84.0.4147.105 Safari/537.36","js":1,"dnt":0,"h":1050,"w":1680,"language":"en-US","geo":{"lat":40.712775,"lon":-74.005973}},"user":{"gender":"M","geo":{"lat":40.712775,"lon":-74.005973},"yob":1982},"ext":{"wrapper":{"wv":"prebid_prebid_4.3.0-pre","transactionId":"dcabc226-9a1c-4433-a0cb-ef43e515e841","wp":"pbjs"}},"source":{"tid":"dcabc226-9a1c-4433-a0cb-ef43e515e841"}}
```

The response returned as a 204 code with no data.  Can you take a look at the above?  If there's anything you'd like me to try, please let me know.

**Comment by pm-shashank-jain** [comment]
@jsnellbaker, thanks for checking with updated adUnits. Please add query param pubmaticTest=true to get the test bid from pubmatic. I've setup a sample code for the same with updated code - >https://codesandbox.io/s/sweet-pare-nykh5?file=/index.html
Test Page -> https://nykh5.csb.app/?pubmaticTest=true 

PS : Please refresh the test page, cause the first time codesandbox transpiles the files. You will the pubmatic ad in player.

File: modules/pubmaticBidAdapter.md [hunk 1]
```diff
@@ -25,6 +25,7 @@ var adUnits = [
       bidder: 'pubmatic',
       params: {
         publisherId: '156209',               // required
+        oustreamAU: 'renderer_test_pubmatic',   // required if mediaTypes-> video-> context is 'outstream'. This value can be get by BlueBillyWig Team.
```
Comment (thekashifmalik) [inline modules/pubmaticBidAdapter.md:28]: This is a typo. It should say `outstreamAU`

**Pull request merged**